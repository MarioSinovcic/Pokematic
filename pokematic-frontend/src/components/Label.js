import React from 'react';
import { Typography, Button } from '@material-ui/core';
import styles from './styledComponents.css';

function Label() {
    const color = "Team Alpha";

    return (
        <div id="Label">
        <Button className="TeamLabel">Lv. 21</Button>
            
        </div>
)
}

export default Label;