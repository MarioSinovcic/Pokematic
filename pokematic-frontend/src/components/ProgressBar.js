import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import LinearProgress from '@material-ui/core/LinearProgress';

const useStyles = makeStyles(theme => ({
  root: {
    width: '100%',
    '& > * + *': {
      marginTop: theme.spacing(2),
    },
    colorPrimary: '#ffffff',
  },
}));

export default function ProgressBar() {
  const classes = useStyles();
  const completed = 30;



  return (
    <div className={classes.root}>
      <LinearProgress variant="determinate" value={completed} color="primary" className="GoalProgressBar"/>
    </div>
  );
}