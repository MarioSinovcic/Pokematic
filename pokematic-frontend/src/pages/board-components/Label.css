.TeamLabel {
    background-color: #F64747 !important;
    color: white !important;
    font-family: 'pkmn_rbygscregular' !important;
    border-radius: 10px !important;
    font-size: 10px !important;
    size: 15px !important;
    margin-bottom: 5px !important;
}