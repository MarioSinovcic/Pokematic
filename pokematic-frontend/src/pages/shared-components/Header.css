.Header{
   width: 100%;
   height: 70px;
   display: inline-flex;
   text-align: center;
   background-color: #f05958;
   box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
   z-index: 3;
   top: 0;
}

.left-icon{
   position: absolute;
   left: 20px;
   top: 13px;
   width: 45px;
   height: 45px;  
   background-repeat: no-repeat;
   background-color: whitesmoke;
   background-size: 30px 30px;
   background-position: center;
   border-radius: 50px;

   transition: 0.5s;
   z-index: 5;
}

.left-icon:hover{
   top: 14px;
   background-color: rgba(225, 225, 225);
   box-shadow: 0px 5px 0px rgba(0, 0, 0, 0.2);
   transition: 0.3s;
}

.middle-section{
   width: 100%;
   text-align: center;
}

.logo{
   width: 250px;
   z-index: 3;
}

.right-icon{
   position: absolute;
   right: 20px;
   top: 13px;
   width: 45px;
   height: 45px; 
   background-repeat: no-repeat;
   background-color: whitesmoke;
   background-size: 30px 30px;
   background-position: center;
   border-radius: 50px;

   transition: 0.5s;
   z-index: 3;
}

.right-icon:hover{
   top: 14px;
   background-color: rgba(225, 225, 225);
   box-shadow: 0px 5px 0px rgba(0, 0, 0, 0.2);
   transition: 0.3s;
}

.profile-image {
    background-image: url('../../images/icons8-instinct-pokemon-96.png');
}

.pokedex-image {
    background-image: url('../../images/icons8-pokeball-96.png');
}

.board-image {
    background-image: url('../../images/icons8-tasks-96.png');
}

