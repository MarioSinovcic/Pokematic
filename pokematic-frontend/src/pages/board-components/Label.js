import React from 'react';
import { Button } from '@material-ui/core';
import './Label.css';

function Label(props) {
    return (
        <div id="Label">
            <Button className="TeamLabel">{props.labelText}</Button>
        </div>
)
}

export default Label;