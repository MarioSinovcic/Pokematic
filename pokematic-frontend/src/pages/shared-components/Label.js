import React from 'react';
import { Button } from '@material-ui/core';
import './Label.css';

function Label(props) {

    return (
        <div id="Label">
            <Button style={ {backgroundColor: props.color} } className="team-label pokemon-label">{props.labelText}</Button>
        </div>
)
}

export default Label;