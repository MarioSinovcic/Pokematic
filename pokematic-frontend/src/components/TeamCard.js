import React from 'react';
import ProgressBar from './ProgressBar';
import { Typography } from '@material-ui/core';
import styles from './styledComponents.css';
import Label from './Label';

function TeamCard() {
    const teamName = "Team Alpha";
    const temporaryTeamLogo = "/eevee.png";

    return (
        <div id="CardShape">
            <div className="BaseShape ShapeContent">
                <div className="TeamIconBG">
                    <img src={temporaryTeamLogo} className="TeamImage"></img>
                </div>
                <div className="TeamStats">
                    <Label />
                    <Typography className="TeamName">{teamName}</Typography>
                    <div className="TeamLevelBar ">
                        <ProgressBar/>
                        <Typography className="EXP EXPText">EXP: 100/200</Typography>
                    </div>
                </div>
            </div>
            <div className="BottomSupport"></div>
            <div className="TriangleCut"></div>
        </div>
)
}

export default TeamCard;