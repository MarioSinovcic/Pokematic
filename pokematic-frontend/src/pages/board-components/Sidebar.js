import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Drawer from '@material-ui/core/Drawer';
import List from '@material-ui/core/List';
import Typography from '@material-ui/core/Typography';
import Divider from '@material-ui/core/Divider';
import ListItem from '@material-ui/core/ListItem';
import ProgressBar from '../../shared-components/ProgressBar';
import './Sidebar.css'

const drawerWidth = 250;

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  appBar: {
    zIndex: theme.zIndex.drawer + 1,
  },
  drawer: {
    width: drawerWidth,
    flexShrink: 0,
  },
  drawerPaper: {
    width: drawerWidth,
  },
  content: {
    flexGrow: 1,
    padding: theme.spacing(3),
  },
  // necessary for content to be below app bar
  toolbar: theme.mixins.toolbar,
}));

export default function Sidebar() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      {/* <CssBaseline />
      <AppBar position="fixed" className={classes.appBar}>
        <Toolbar>
          <Typography variant="h6" noWrap>
            Pokematic
          </Typography>
        </Toolbar>
      </AppBar> */}
      <Drawer
        className="SideBar"
        variant="permanent"
        classes={{
          paper: classes.drawerPaper,
        }}
      >
        <div className={classes.toolbar} />
        <List>
         
            <ListItem button key={"All Tasks"} className="TeamTabs TaskButton">
            <Typography className="TaskFilter AllTasks">ALL TASKS</Typography>
            </ListItem>

             <ListItem button key={"My Tasks"} className="TeamTabs">
             <Typography className="TaskFilter MyTasks">MY TASKS</Typography>
            </ListItem>
        </List>
        <Divider className="SideBarDivider"/>
        <List>
          {/* Dynamically fetch goals here */}
          {['Planning', 'Frontend Team', 'Testers', 'Design Squad'].map((text, index) => (
            <div className="TeamTabs">
            <ListItem button key={text} className="TeamTabs">
              <Typography className="TaskName">{text}</Typography>
              <ProgressBar />
            </ListItem>
            <Divider className="GoalDivider" /> 
            </div>
          ))}
        </List>
      </Drawer>
    </div>
  );
}