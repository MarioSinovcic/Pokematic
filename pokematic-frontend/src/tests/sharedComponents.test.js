import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import rootReducers from '../reducers';
import TeamCard from '../shared-components/TeamCard';
import Header from '../shared-components/Header';
import ModalButton from '../shared-components/ModalButton';
import ProgressBar from '../shared-components/ProgressBar';
import Sidebar from '../shared-components/Sidebar';
import TeamDetails from '../shared-components/TeamDetails';
import Label from '../shared-components/Label';


describe('shared components', () => {

    let store;
    beforeEach(() => {
        store = createStore(rootReducers)
    })


    describe('snapshot matching', () => {

        it('Team Card matches snapshot' , () => {
            const teamCard = renderer.create(<Provider store={store}>
                <TeamCard />
            </Provider>).toJSON();
            expect(teamCard).toMatchSnapshot();
        });

        // TODO: figure out how to use snapshot for router
        xit('Header matches snapshot' , () => {
            const header = renderer.create(<Provider store={store}>
                <Header />
            </Provider>).toJSON();
            expect(header).toMatchSnapshot();
        });

        it('Modal Button matches snapshot' , () => {
            const modalButton = renderer.create(<Provider store={store}>
                <ModalButton />
            </Provider>).toJSON();
            expect(modalButton).toMatchSnapshot();
        });

        it('Progress Bar matches snapshot' , () => {
            const progressBar = renderer.create(<Provider store={store}>
                <ProgressBar />
            </Provider>).toJSON();
            expect(progressBar).toMatchSnapshot();
        });

        it('Sidebar matches snapshot' , () => {
            const dummyTeamsList = [];

            const sidebar = renderer.create(<Provider store={store}>
                <Sidebar 
                teamsList={dummyTeamsList}/>
            </Provider>).toJSON();
            expect(sidebar).toMatchSnapshot();
        });

        it('Team Details matches snapshot' , () => {
            const details = renderer.create(<Provider store={store}>
                <TeamDetails />
            </Provider>).toJSON();
            expect(details).toMatchSnapshot();
        });

        it('Label matches snapshot' , () => {
            const label = renderer.create(<Provider store={store}>
                <Label />
            </Provider>).toJSON();
            expect(label).toMatchSnapshot();
        });


    })


    




});