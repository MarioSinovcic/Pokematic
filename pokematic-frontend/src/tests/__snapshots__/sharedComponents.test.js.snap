// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared components snapshot matching Label matches snapshot 1`] = `
<div
  id="Label"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text team-label pokemon-label"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "backgroundColor": undefined,
      }
    }
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    />
  </button>
</div>
`;

exports[`shared components snapshot matching Modal Button matches snapshot 1`] = `
<div>
  <div>
    <button
      aria-label="add"
      className="MuiButtonBase-root MuiFab-root dark-theme-bg MuiFab-secondary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiFab-label"
      />
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <div />
</div>
`;

exports[`shared components snapshot matching Progress Bar matches snapshot 1`] = `
<div
  className="progress-bar"
>
  <div
    className="MuiLinearProgress-root MuiLinearProgress-colorPrimary goal-progress-bar MuiLinearProgress-determinate"
    role="progressbar"
  >
    <div
      className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
      style={Object {}}
    />
  </div>
</div>
`;

exports[`shared components snapshot matching Sidebar matches snapshot 1`] = `
<div
  className="makeStyles-root-113"
>
  <div
    className="MuiDrawer-root MuiDrawer-docked SideBar"
  >
    <div
      className="MuiPaper-root MuiDrawer-paper makeStyles-drawerPaper-116 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
    >
      <div
        className="makeStyles-toolbar-118"
      />
      <ul
        className="MuiList-root MuiList-padding"
      >
        <div
          aria-disabled={false}
          className="MuiButtonBase-root MuiListItem-root TeamTabs TaskButton MuiListItem-gutters MuiListItem-button"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <p
            className="MuiTypography-root TaskFilter AllTasks MuiTypography-body1"
          />
        </div>
      </ul>
      <hr
        className="MuiDivider-root SideBarDivider"
      />
      <ul
        className="MuiList-root MuiList-padding"
      />
    </div>
  </div>
</div>
`;

exports[`shared components snapshot matching Team Card matches snapshot 1`] = `
<div
  id="CardShape"
>
  <div
    className="base-shape shape-content"
  >
    <div>
      <div
        className="team-details"
      >
        <div
          className="team-icon-bg"
        >
          <img
            alt="team-logo"
            className="team-image"
          />
        </div>
        <div
          className="team-stats"
        >
          <div
            id="Label"
          >
            <button
              className="MuiButtonBase-root MuiButton-root MuiButton-text team-label pokemon-label"
              disabled={false}
              onBlur={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              style={
                Object {
                  "backgroundColor": "#F64747",
                }
              }
              tabIndex={0}
              type="button"
            >
              <span
                className="MuiButton-label"
              >
                lv. undefined
              </span>
            </button>
          </div>
          <p
            className="MuiTypography-root team-name MuiTypography-body1"
          />
          <div
            className="team-level-bar card-width"
          >
            <div
              className="progress-bar"
            >
              <div
                aria-valuenow={NaN}
                className="MuiLinearProgress-root MuiLinearProgress-colorPrimary goal-progress-bar MuiLinearProgress-determinate"
                role="progressbar"
              >
                <div
                  className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
                  style={
                    Object {
                      "transform": "translateX(NaN%)",
                    }
                  }
                />
              </div>
            </div>
            <p
              className="MuiTypography-root EXP EXP-text MuiTypography-body1"
            >
              EXP: 
              / 
              NaN
            </p>
          </div>
        </div>
      </div>
      <hr
        className="MuiDivider-root GoalDivider"
      />
    </div>
  </div>
  <div
    className="bottom-support"
  />
  <div
    className="triangle-cut"
  />
</div>
`;

exports[`shared components snapshot matching Team Details matches snapshot 1`] = `
<div>
  <div
    className="team-details"
  >
    <div
      className="team-icon-bg"
    >
      <img
        alt="team-logo"
        className="team-image"
      />
    </div>
    <div
      className="team-stats"
    >
      <div
        id="Label"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text team-label pokemon-label"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "backgroundColor": "#F64747",
            }
          }
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            lv. undefined
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <p
        className="MuiTypography-root team-name MuiTypography-body1"
      />
      <div
        className="team-level-bar card-width"
      >
        <div
          className="progress-bar"
        >
          <div
            aria-valuenow={NaN}
            className="MuiLinearProgress-root MuiLinearProgress-colorPrimary goal-progress-bar MuiLinearProgress-determinate"
            role="progressbar"
          >
            <div
              className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
              style={
                Object {
                  "transform": "translateX(NaN%)",
                }
              }
            />
          </div>
        </div>
        <p
          className="MuiTypography-root EXP EXP-text MuiTypography-body1"
        >
          EXP: 
          / 
          NaN
        </p>
      </div>
    </div>
  </div>
  <hr
    className="MuiDivider-root GoalDivider"
  />
</div>
`;
